Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -86,4 +91,96 @@ namespace ROS2
         return joints;
     }
 
+    AZStd::unordered_set<AZStd::string> Utils::GetMeshesFilenames(const urdf::LinkConstSharedPtr& root_link, bool visual, bool colliders)
+    {
+        AZStd::unordered_set<AZStd::string> filenames;
+        const auto getFnFromGeometry = [](const urdf::GeometrySharedPtr& geometry)
+        {
+            if (geometry->type == urdf::Geometry::MESH)
+            {
+                auto pMesh = std::dynamic_pointer_cast<urdf::Mesh>(geometry);
+                if (!pMesh)
+                {
+                    return AZStd::string{};
+                }
+                return AZStd::string(pMesh->filename.c_str(), pMesh->filename.size());
+            }
+            return AZStd::string{};
+        };
+
+        const auto processLink = [&](const urdf::Link& link)
+        {
+            if (visual)
+            {
+                for (auto& p : link.visual_array)
+                {
+                    const auto fn = getFnFromGeometry(p->geometry);
+                    if (!fn.empty())
+                    {
+                        filenames.insert(fn);
+                    }
+                }
+            }
+            if (colliders)
+            {
+                for (auto& p : link.collision_array)
+                {
+                    const auto fn = getFnFromGeometry(p->geometry);
+                    if (!fn.empty())
+                    {
+                        filenames.insert(fn);
+                    }
+                }
+            }
+        };
+
+        std::function<void(const std::vector<urdf::LinkSharedPtr>&)> link_visitor =
+            [&](const std::vector<urdf::LinkSharedPtr>& child_links) -> void
+        {
+            for (auto child_link : child_links)
+            {
+                processLink(*child_link);
+                link_visitor(child_link->child_links);
+            }
+        };
+        processLink(*root_link);
+        link_visitor(root_link->child_links);
+        return filenames;
+    }
+
+    /// Finds global path from URDF path
+    AZStd::string Utils::ResolveURDFPath(
+        AZStd::string path, const AZStd::string& urdf_path, const AZStd::function<bool(const AZStd::string&)>& fileExists)
```

Comment:
naming conventions - check the entire file (camel case). Also, apply other standards.